### PR TITLE
Added Social Security Number validator

### PR DESCRIPTION
**Pull request by kishanmodi**

### CODE REVIEW

**Comment by kishanmodi** [comment]
solved conflict

File: types.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const EmailAddress = require('./validators/email');
 const PAN_IN = require('./validators/');
 const AadhaarNumber = require('./validators/Aadhaar_IN.js');
 const DrivingLC_IN = require('./validators/DRIVING_LC_IN');
-const SSN = require('./validators/ssn.js');
+const SSN = require('./validators/ssn_us.js');
```
Comment (SwapnilSoni1999) [inline types.js]: make uppercase and rename variable to `SSN_US`

File: types.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 const ObjectId = require('./validators/objectid');
 const uuid = require('./validators/uuid');
 const EmailAddress = require('./validators/email');
-const PAN_IN = require('./validators/PAN_IN');
-const AadhaarNumber = require('./validators/Aadhaar_IN');
+const PAN_IN = require('./validators/');
+const AadhaarNumber = require('./validators/Aadhaar_IN.js');
```
Comment (SwapnilSoni1999) [inline types.js]: remove `.js` from here

File: validators/ssn_us.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const dry = require('drytypes')
+
+module.exports = dry.makeDryType((x) => {
+    if (typeof x !== 'string') {
+        return { success: false, message: 'Social Security Number should be string!' }
+    }
+
+    let ssnRegex = /^(?!666|000|9\\d{2})\\d{3}-(?!00)\\d{2}-(?!0{4})\\d{4}$/;
```
Comment (SwapnilSoni1999) [inline validators/ssn_us.js]: regex is built once, use `const`

File: validators/ssn_us.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    if (!x.match(ssnRegex)) {
+        return { success: false, message: 'Invalid SSN!, Please Recheck the SSN!' }
+    }
+
+    return { success: true, message:'You Have Entered Valid Social Security Number!' }
+
+}, 'SSN')
```
Comment (SwapnilSoni1999) [inline validators/ssn_us.js]: `SSN_US` is needed here

**Review by SwapnilSoni1999 — CHANGES_REQUESTED** [review]
Please make required changes

**Review by SwapnilSoni1999 — APPROVED** [review]

**Pull request merged**